lib/multiplexer/proxyline.go

```diff
@@ -122,3 +127,72 @@ func parseIP(protocol string, addrString string) (net.IP, error) {
 	}
 	return addr, nil
 }
+
+type proxyV2Header struct {
+	Signature      [12]uint8
+	VersionCommand uint8
+	Protocol       uint8
+	Length         uint16
+}
+
+type proxyV2Address4 struct {
+	Source          [4]uint8
+	Destination     [4]uint8
+	SourcePort      uint16
+	DestinationPort uint16
+}
+
+type proxyV2Address6 struct {
+	Source          [16]uint8
+	Destination     [16]uint8
+	SourcePort      uint16
+	DestinationPort uint16
+}
+
+func ReadProxyLineV2(reader *bufio.Reader) (*ProxyLine, error) {
+	var header proxyV2Header
+	var ret ProxyLine
+	if err := binary.Read(reader, binary.BigEndian, &header); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if bytes.Compare(header.Signature[:], proxyV2Prefix) != 0 {
+		return nil, trace.BadParameter("unrecognized signature %s", hex.EncodeToString(header.Signature[:]))
+	}
+	cmd, ver := header.VersionCommand&0xF, header.VersionCommand>>4
+	if ver != 2 {
```

Comment:
Can we add constants for these magic numbers?
- version 2
- cmd 0
- cmd 1
- protocol 0x11
- protocol 0x21

Reply:
Added constants